Resources/public/js/views/dashboard/ez-dashboardblockallcontentview.js

```diff
@@ -0,0 +1,157 @@
+
+    var BLOCK_IDENTIFIER = 'all-content',
+        CLASS_LOADING = 'is-loading',
+        CLASS_ROW_SELECTED = 'is-row-selected',
+        SELECTOR_ROW = '.ez-allcontent-block-row',
+        SELECTOR_CONTENT = '.ez-allcontent-block-content',
+        EVENTS = {};
+
+    EVENTS[SELECTOR_ROW] = {tap: '_uiShowRowOptions'};
+
+    /**
+     * The all content dashboard block view
+     *
+     * @namespace eZ
+     * @class DashboardBlockAllContentView
+     * @constructor
+     * @extends eZ.DashboardBlockBaseView
+     */
+    Y.eZ.DashboardBlockAllContentView = Y.Base.create('dashboardBlockAllContentView', Y.eZ.DashboardBlockBaseView, [Y.eZ.AsynchronousView], {
+        initializer: function () {
+            this._fireMethod = this._getContentData;
+            /**
+             * Stores the click outside event handler
+             *
+             * @property _clickOutsideHandler
+             * @type {Object}
+             * @protected
+             */
+            this._clickOutsideHandler = null;
+            this.events = Y.merge(this.events, EVENTS);
+
+            this._set('identifier', BLOCK_IDENTIFIER);
+            this.get('container').addClass(this._generateViewClassName(Y.eZ.DashboardBlockBaseView.NAME));
+        },
+
+        /**
+         * Renders the dashboard block view
+         *
+         * @method render
+         * @return {eZ.DashboardBlockAllContentView} the view itself
+         */
+        render: function () {
+            this.get('container').setHTML(this.template()).addClass(CLASS_LOADING);
+
+            return this;
+        },
+
+        /**
+         * Shows the row options
+         *
+         * @method _uiShowOptions
+         * @protected
+         * @param event {Object} event facade
+         */
+        _uiShowRowOptions: function (event) {
+            this._uiHideRowOptions();
+
+            event.currentTarget.addClass(CLASS_ROW_SELECTED);
+
+            this._clickOutsideHandler = this.get('container').one(SELECTOR_CONTENT).on('clickoutside', Y.bind(this._uiHideRowOptions, this));
+        },
+
+        /**
+         * Clears the click outside handler
+         *
+         * @method _clearClickOutsideHandler
+         * @protected
+         */
+        _clearClickOutsideHandler: function () {
+            if (this._clickOutsideHandler) {
+                this._clickOutsideHandler.detach();
+            }
+        },
+
+        /**
+         * Hides the row options
+         *
+         * @method _uiHideOptions
+         * @protected
+         */
+        _uiHideRowOptions: function () {
+            this.get('container').all(SELECTOR_ROW).removeClass(CLASS_ROW_SELECTED);
+
+            this._clearClickOutsideHandler();
+
+        },
+
+        /**
+         * Makes request for data
+         *
+         * @method _getContentData
+         * @protected
+         * @param event {Object} event facade
+         */
+        _getContentData: function () {
+            this.fire('contentSearch', {
```

Comment:
given your criteria is on the subtree, you should rather do a Location search. That way, you get a list of Location in return (which embeds the Content Info so you have everything in hand), this will avoid to use the main Location while displaying the result since this main Location is maybe not the correct one regarding that criteria.

Also, using the `locationSearch` allows to configure an attribute (instead of the `callback` a few lines below) that will receive the search results to fully benefit from the asynchronous view extension. With that, you also have the error handling for *free* (which is missing at the moment).